lib/filter-blocked.js

```diff
@@ -1,25 +1,54 @@
 const async = require('async')
+const pull = require('pull-stream')
 const pullFilterAsync = require('./pull-filter-async')
+const paramap = require('pull-paramap')
 
 module.exports = function (blockSources, { isBlocking, checkRoot = false, useRootAuthorBlocks = false }) {
-  return pullFilterAsync((msg, cb) => {
-    var checks = []
-    blockSources.forEach(source => {
-      if (msg.value) {
-        checks.push({ source, dest: msg.value.author })
-      }
-      if (checkRoot && msg.root && msg.root.value) {
-        checks.push({ source, dest: msg.root.value.author })
+  // Maybe hacky but it seems true in practice
+  const me = blockSources[0]
+  return pull(
+    paramap((msg, cb) => {
+      var checks = []
+
+      // It doesn't make sense to have the forEach here if blockSources would
+      // only contain you and the thread author, but I think this code is in
+      // preparation for block lists, in which case it makes sense
+      blockSources.forEach(source => {
+        if (msg.value) {
+          checks.push({ source, dest: msg.value.author })
+        }
+        if (checkRoot && msg.root && msg.root.value) {
+          checks.push({ source, dest: msg.root.value.author })
+        }
+      })
+
+      if (useRootAuthorBlocks && msg.root && msg.root.value) {
+        checks.push({ source: msg.root.value.author, dest: msg.value.author })
       }
-    })
 
-    if (useRootAuthorBlocks && msg.root && msg.root.value) {
-      checks.push({ source: msg.root.value.author, dest: msg.value.author })
-    }
+      async.map(checks, isBlocking, (err, res) => {
+        if (err) return cb(err)
 
-    async.any(checks, isBlocking, (err, blocking) => {
-      if (err) return cb(err)
-      cb(null, !blocking)
-    })
-  }, 10)
+        // Important that we start with 0 (us), we're prioritized above thread
+        // authors
+        for (let i = 0; i < checks.length; i++) {
+          if (res[i]) {
+            // I shouldn't be modifying the original msg in a (para)map I guess
+            // but :shrug: (? or will I break stuff)
```

Comment:
You can use `msg.value.meta` for this, see also: https://github.com/ssbc/ssb-db/pull/228#issuecomment-435438060

I'd recommend `_.get()` and `_.set()`, for example:

```js
const _ = require('lodash')

_.set(msg, 'value.meta.isBlocked', true)
```

This way you can also avoid this sort of thing:

```js
msg.value.meta && msg.value.meta.blockedBy && msg.value.meta.blockedBy.id

// vs

_.get(msg, 'value.meta.blockedBy.id')
```
```